kernel-api/src/main/scala/org/apache/toree/dependencies/CoursierDependencyDownloader.scala

```diff
@@ -76,15 +76,16 @@ class CoursierDependencyDownloader extends DependencyDownloader {
     trace: Boolean,
     configuration: Option[String] = None,
     artifactType: Option[String] = None,
-    artifactClassifier: Option[String] = None
+    artifactClassifier: Option[String] = None,
+    excludes: Set[(String,String)] = Set.empty
```

Comment:
As you are adding a new parameter here, you also need to update the addDepsSpec where it defines a mock of a dependency downloader and add a new parameter to the mock (any[Set[(String,String)]]). I am playing with it and if that's the only thing I might update your pr before merging, otherwise please make sure you look into this.